src/Core__Option.res

```diff
@@ -34,10 +34,16 @@ let forEach = (opt, f) =>
   | None => ()
   }
 
-let getExn = x =>
+let getExn = (x, ~message=?) =>
   switch x {
   | Some(x) => x
-  | None => raise(Not_found)
+  | None =>
+    Core__Error.panic(
+      switch message {
+      | None => "Not found."
```

Comment:
Looking at the implementation of `panic`, this means that by default an Error will be raised with the message "Panic! Not found." which sounds a bit weird to me.

Not sure what would be a better message, maybe something like
"Panic! Option.getExn called for None value"
?

Reply:
Yeah, better. Changing.

Reply:
What about using a custom exception? It could be in a different function though.